pkg/apis/pipeline/v1alpha1/build_gcs_resource.go

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2019 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"flag"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	buildGCSFetcherImage = flag.String("gcs-fetcher-image", "gcr.io/cloud-builders/gcs-fetcher:latest",
+		"The container image containing our GCS fetcher binary.")
+	buildGCSUploaderImage = flag.String("gcs-uploader-image", "gcr.io/cloud-builders/gcs-uploader:latest",
+		"The container image containing our GCS uploader binary.")
+)
+
+// GCSArtifactType defines a type of GCS resource.
+type GCSArtifactType string
+
+const (
+	// GCSArchive indicates that resource should be fetched from a typical archive file.
+	GCSArchive GCSArtifactType = "Archive"
+
+	// GCSManifest indicates that resource should be fetched using a
+	// manifest-based protocol which enables incremental source upload.
+	GCSManifest GCSArtifactType = "Manifest"
+)
+
+// BuildGCSResource describes a resource in the form of an archive,
+// or a source manifest describing files to fetch.
+type BuildGCSResource struct {
+	Name           string
+	Type           PipelineResourceType
+	Location       string
+	DestinationDir string
+	ArtifactType   GCSArtifactType
+	//Secret holds a struct to indicate a field name and corresponding secret name to populate it
+	Secrets []SecretParam `json:"secrets"`
```

Comment:
(don't need the `json` notation here, we should remove that from the other resource objects too when we get a chance :innocent: )